### ISSUE TITLE
6.1.3.3	Update statuses as part of delivery of service for requests as per functionality in 6.1.4

### ISSUE THREAD

**Issue by MKSessions** [HTBox/allReady#703]
As requests are scheduled, completed, or marked incomplete track that status in the request database.

Issue 6.1.4 is #697

**Comment by MisterJames** [comment]
We are working with getasmokealarm folks to detail out how we will keep these status markers in check. They are likely going to need to add a flag to their endpoint that allows us to mark a request as "managed by third party".  I will update this as we go.

**Comment by stevejgordon** [comment]
@tonysurma -  I have functionality within the site for org admins to mark requests as completed from the itinerary management page. Completed requests can then be marked incomplete to reset them to assigned. This is stored against the request in our database.

The only part missing I think would be a method to allow admins to mark a request cancelled and provide a cancellation reason.

Is there anything else missing? 

Once confirmation, I can pick this up and provide a means to mark requests as cancelled if you want to assign to me as one of my next tasks.

**Comment by stevejgordon** [comment]
@MisterJames @tonysurma - As we have already the code to handle marking a request completed (and reverting back to incomplete) I think this could fairly easily be updated to call an endpoint with the request details and new status. Do we have a test endpoint and token/auth mechanism we can work against?

On a work project we're using the Polly library (https://github.com/App-vNext/Polly/) to manage retry policies for HttpClient requests between REST services. This seems to work quite well so might be good to include here.

We have the GUID RequestId in the model at the moment. I presume this originally comes from getasmokealarm and would be the ID we send back along with an updated status for them to store?

**Comment by tonysurma** [comment]
@MisterJames @stevejgordon what is the status on this?

**Comment by stevejgordon** [comment]
@tonysurma - TLDR - I think this can be closed.

@MisterJames and I chatted for a little while after the standup and I have been able to get a better grasp of the integration piece. For the above issue specifically I believe we can consider this closed. The requirement here (if I understand it correctly) is to ensure that we have the concept of a status for a request and a way to manage and store changes through the request's lifetime.

We currently have that in place and while it could be enhanced I believe for v1 it's probably enough. Requests are initially "unassigned". When added to an itinerary they are "assigned". From there they can be marked as "cancelled" or "completed". This is currently all org admin behaviour within the event and itinerary pages.

I think the requirement to update GASA is covered in issue #704. @MisterJames has forwarded me some API specs which look to cover how we can send an update to the GASA site when a request changes status. I can pick that up once some of the geocoding work is completed.
